reader_concurrency_semaphore.hh

```diff
@@ -195,12 +206,20 @@ public:
 
     reader_concurrency_semaphore(int count,
             ssize_t memory,
+            sstring name = "",
```

Comment:
Let's make this mandatory (no default) and remove all the special logic for a `""` name.

Reply:
Done